packages/babel-plugin-minify-simplify/src/index.js

```diff
@@ -28,6 +28,62 @@ module.exports = ({ types: t }) => {
   const or = (a, b) => t.logicalExpression("||", a, b);
   const and = (a, b) => t.logicalExpression("&&", a, b);
 
+  const operators = new Set([
+    '+', '-', '*', '%',
+    '<<', '>>', '>>>',
+    '&', '|', '^', '/',
+    '**'
+  ]);
+
+  const updateOperators = new Set([
+    '+', '-'
+  ]);
+
+  function isEqual(arr1, arr2) {
+    return arr1.every((value, index) => {
+      return value.toString() === arr2[index].toString();
+    });
+  }
+
+  function getName(node) {
+    if (node.type === 'ThisExpression') {
+      return 'this';
+    }
+    if (node.type === 'Super') {
+      return 'super';
+    }
+    // augment identifiers so that they don't match
+    // string/number literals
+    // but still match against each other
+    return node.name
+      ? node.name + '_'
+    	: node.value /* Literal */;
+  }
+
+  function getPropNames(path) {
+    if (!path.isMemberExpression()) {
+      return;
+    }
+
+    let obj = path.get('object');
+
+    const prop = path.get('property');
+    const propNames = [getName(prop.node)];
+
+    while (obj.type !== 'Identifier' &&
+           obj.type !== 'ThisExpression' &&
+           obj.type !== 'Super') {
+      const node = obj.get('property').node;
+      if (node) {
+        propNames.push(getName(node));
+      }
+      obj = obj.get('object');
```

Comment:
nit: `.get()` calls are more expensive than just directly referencing the nodes' children.

It's probably faster/clearer to use nodes instead of `.get()`, which returns paths. This doesn't need path-specific methods, like scoping.